### PR TITLE
executor: migrate test-infra to testify for executor_required_rows_test.go

### LINKED ISSUE #28576: migrate test-infra to testify for `executor_required_rows_test.go`

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #28576 
### Release note

```release-note
None
```

### CODE REVIEW

File: executor/executor_required_rows_test.go [hunk 1]
```diff
@@ -190,15 +182,15 @@ func (s *testExecSuite) TestLimitRequiredRows(c *C) {
 		ctx := context.Background()
 		ds := newRequiredRowsDataSource(sctx, testCase.totalRows, testCase.expectedRowsDS)
 		exec := buildLimitExec(sctx, ds, testCase.limitOffset, testCase.limitCount)
-		c.Assert(exec.Open(ctx), IsNil)
+		require.Nil(t, exec.Open(ctx))
```
Comment (tisonkun) [inline executor/executor_required_rows_test.go:185]:
```suggestion
		require.NoError(t, exec.Open(ctx))
```

ditto others

**Review by tisonkun — COMMENTED** [review]
Rest LGTM.

File: executor/executor_required_rows_test.go [hunk 2]
```diff
@@ -900,37 +892,37 @@ func (s *testExecSuite) TestVecGroupCheckerDATARACE(c *C) {
...
+			require.Equal(t, "123", vgc.lastRowDatums[0].GetMysqlDecimal().String())
 		case mysql.TypeJSON:
-			c.Assert(vgc.firstRowDatums[0].GetMysqlJSON().String(), Equals, `{"123": 123}`)
-			c.Assert(vgc.lastRowDatums[0].GetMysqlJSON().String(), Equals, `{"123": 123}`)
+			require.Equal(t, `{"123": 123}`, vgc.firstRowDatums[0].GetMysqlJSON().String())
+			require.Equal(t, `{"123": 123}`, vgc.lastRowDatums[0].GetMysqlJSON().String())
 			chk.Column(0).ReserveJSON(1)
 			j := new(json.BinaryJSON)
-			c.Assert(j.UnmarshalJSON([]byte(fmt.Sprintf(`{"%v":%v}`, 456, 456))), IsNil)
+			require.Nil(t, j.UnmarshalJSON([]byte(fmt.Sprintf(`{"%v":%v}`, 456, 456))))
```
Comment (xhebox) [inline executor/executor_required_rows_test.go:922]: This line should be `NoError`, too.

Comment (tisonkun) [reply executor/executor_required_rows_test.go]: I've submit a patch to fix all of this for @feitian124 :)

**Review by tisonkun — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

File: executor/executor_required_rows_test.go (hunk 2 above)
Comment (feitian124) [reply executor/executor_required_rows_test.go]: thanks

**Pull request merged**